proposal/custom-url-resolution.md

```diff
@@ -0,0 +1,230 @@
+# Custom url resolution: Draft 1
+
+_[(Issue)](https://github.com/sass/sass/issues/2535)_
+
+## Table of Contents
+
+- [Background](#background)
+- [Summary](#summary)
+  - [Example](#example)
+- [Function](#function)
+  - [Steps](#steps)
+  - [JavaScript API](#javascript-api)
+- [Presets](#presets)
+  - [The inline Preset](#the-inline-preset)
+    - [Inline Preset Usage](#inline-preset-usage)
+    - [Inline Preset Implementation](#inline-preset-implementation)
+  - [The relative Preset](#the-relative-preset)
+    - [Relative Preset Usage](#relative-preset-usage)
+    - [Relative Preset Implementation](#relative-preset-implementation)
+
+## Background
+
+> This section is non-normative.
+
+Many css features require the use of a sass-url reference to reference resources from outside the sass files, however these files also need to exist on the eventual output directory and server. To ensure the references are valid, the sass API should allow for the user to provide a way to remap and/or inline these resources.
```

Comment:
Please use proper capitalization: "CSS", "Sass", etc. Also, code-quote any use of Sass or CSS identifiers or syntax such as `sass-url`.